### ISSUE TITLE
Heading is not spec compliant: missing role="heading" element

### ISSUE THREAD

**Issue by ryami333** [springload/react-accessible-accordion#166]
The [aria spec](https://www.w3.org/TR/wai-aria-practices-1.1/#accordion) includes the following:

> Each accordion header button is wrapped in an element with role heading that has a value set for aria-level that is appropriate for the information architecture of the page.

We are currently only including the 'button' part of the markup, but not the 'heading' part. The way I see it, we have two options:

- Either, render a `<div role="heading" aria-level="x>` around the button in `AccorionItemHeading`.
- Or, maintain separate `AccordionItemHeading` and `AccordionItemButton` components.

The first option is made complicated by the `className` and `expandedClassName` props. Would we have to rename these to `headingClassName`, ~`headingExpandedClassName`~, `buttonClassName` and ~`buttonExpandedClassName`~ props?? That's really verbose and messy, IMO.

> EDIT: we no longer need `expandedClassName` props, since this edge case can be handled with `AccordionItemState` (and it _is_ and edge case now that we are applying the `hidden` attribute to panels.

The second option solves the className problem, but it exposes the user to a few markup pitfalls which have the potential of making their accordion non-compliant. Eg:

```jsx
/**
 * COPY + PASTE WARNING!!!
 * This is pseudo-code. Copy and pasting it into your application will not work.
 */

// valid ✅
function Foo() {
  return (
    <AccordionItemHeading>
      <AccordionItemButton>New Zealand</AccordionItemButton>
    </AccordionItemHeading>
  );
}

// invalid ❌
function Bar() {
  return (
    <AccordionItemHeading>
      New Zealand
      <AccordionItemButton>Click me</AccordionItemButton>
    </AccordionItemHeading>
  );
}
```

We can mitigate these risks with `propType` validation. Still - is it more or less ergonomic?

**Comment by ryami333** [comment]
Now that I've [implemented](https://github.com/springload/react-accessible-accordion/pull/191) the first option (both 'heading' and 'button' consolidated into `AccordionItemHeading`) it feels even more awkward than I originally thought 😞 

Now I'm leaning more toward the second option. It feels the more 'correct' option, or the most consistent at the very least. We can use 'prop-types', and potentially also forward refs to warn users against invalid markup to mitigate some of the pitfalls associated with that approach. I might try it in  PR and see how it _feels_.

**Comment by ryami333** [comment]
Has been addressed in `next` branch, will be published in `3.0.0` 🍾
